### PR TITLE
Bluetooth: ISO: Proper handling of sending data for unidirectional channels

### LINKED ISSUE #41946: Bluetooth: ISO: Sending on RX-only CIS doesn't report error
**Describe the bug**
If there's a unidirectional CIS connected where Zephyr can receive only, then when calling `bt_iso_chan_send()` we don't receive any errors, and the `sent` callback is called almost immediately. 

**To Reproduce**
Use `iso_connected_benchmark` sample to setup a unidirectional CIS (TX-only) on the central and notice no errors when the peripheral sends. 

**Expected behavior**
`bt_iso_chan_send()` should return an error if the device cannot send. The application should be able to determine whether or not it can send before calling the function as well. 

**Impact**
Annoyance. 

**Logs and console output**
N/A

**Environment (please complete the following information):**
 - Commit SHA or Version used: e8df8a579033be238beb774e4909d637da649f3a

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Thalley**
This PR ensures that no data is sent to the controller or remote device from the host if the ISO channel is a unidirectional RX-only channel. 
This is achieved by add a check in `bt_iso_chan_send` to return an error if the channel cannot send, as well as exposing that value through the `bt_iso_get_info` so that the application can verify the information before it attempts to send. 

depends on https://github.com/zephyrproject-rtos/zephyr/pull/42034
fixes https://github.com/zephyrproject-rtos/zephyr/issues/41946

### CODE REVIEW

**Comment by Thalley** [comment]
ping @erikrobstad

**Review by asbjornsabo — DISMISSED** [review]
LGTM
Very nicely structured into commits - easy to review :-)

**Review by jhedberg — DISMISSED** [review]

**Review by asbjornsabo — DISMISSED** [review]

**Review by asbjornsabo — DISMISSED** [review]

**Review by Casper-Bonde-Bose — COMMENTED** [review]
I appreciate the can_send and can_receive booleans are easy to understand, but as we also use sink/source for the same thing I'm afraid it will be confusing. It there a specific reason not to use the existing sink/source information here?
(or maybe source/sink is introduced in another PR?)

**Comment by Thalley** [comment]
> I appreciate the can_send and can_receive booleans are easy to understand, but as we also use sink/source for the same thing I'm afraid it will be confusing. It there a specific reason not to use the existing sink/source information here? (or maybe source/sink is introduced in another PR?)

Can you elaborate on what you mean by using sink/source here?

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Comment by Casper-Bonde-Bose** [comment]
> > I appreciate the can_send and can_receive booleans are easy to understand, but as we also use sink/source for the same thing I'm afraid it will be confusing. It there a specific reason not to use the existing sink/source information here? (or maybe source/sink is introduced in another PR?)
> 
> Can you elaborate on what you mean by using sink/source here?

A sink stream is receiving data where a source stream is transmitting data. For example in bt_audio_pac_type we use the term source/sink like in the specifications and also the majority of the API's use src/sink. But as this is on the ISO layer it might be OK to use transmit and receive terms in stead. It also corresponds to the tx/rx qos terms used in the ISO layer already. I rest my case - let's stick to existing naming :-)

**Review by Casper-Bonde-Bose — DISMISSED** [review]
LGTM

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Comment by Thalley** [comment]
Rebased to solve merge conflict

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Review by Casper-Bonde-Bose — APPROVED** [review]
Re-Approve after rebase

**Review by rick1082 — APPROVED** [review]

**Review by asbjornsabo — APPROVED** [review]

**Comment by Thalley** [comment]
> > depends on #42034
> 
> This is merged now; do you need to rebase this one?

Does it need a rebase? 

But if so, then sure, would like to get this merged as well.

**Comment by mbolivar-nordic** [comment]
> Does it need a rebase?
> 
> But if so, then sure, would like to get this merged as well.

I'm asking because I don't understand the dependency -- usually if PR 1 depends on PR 2, then after 2 is merged, 1 must be rebased. I'm asking you if you need to do that, or if this is OK as-is

**Comment by Thalley** [comment]
> > Does it need a rebase?
> > But if so, then sure, would like to get this merged as well.
> 
> I'm asking because I don't understand the dependency -- usually if PR 1 depends on PR 2, then after 2 is merged, 1 must be rebased. I'm asking you if you need to do that, or if this is OK as-is

Ah, I see. The PR that it depended on was merged back in January and has been rebased since then, so it should be alright. Should/could probably have removed the dependency :) 

It should be fine to merge

**Pull request merged**
